SmartDeviceLink/SDLIAPSession.m

```diff
@@ -139,6 +276,37 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession) {
+            NSError *sendErr = nil;
+            
+            if (![strongSelf sdl_dequeueAndWriteToOutputStream:&sendErr] && sendErr != nil) {
+                [strongSelf sdl_handleOutputStreamWriteError:sendErr];
+            }
+        }
+    };
+}
+
+#pragma mark - Lifecycle Destruction
+
+- (void)dealloc {
+    [self.sendDataQueue flush];
```

Comment:
OK, then I would propose putting the flush call into a dealloc method for the queue because the session could go away with multiple buffers enqueued. The buffers will be held by the NSMutableArray and therefore the array may not go away, leading to the queue not going away and a leak.